### PR TITLE
Handle updates to bosh-linux-stemcell-builder/ci/docker/...

### PR DESCRIPTION
**Pull request by aramprice**
> Blocked on https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/372

After https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/372 there will no longer be a separate
ci/docker/os-image-stemcell-builder-jammy directory, and instead only be one single `ci/docker/os-image-stemcell-builder/` directory _in each branch, for that branch_ of [cloudfoundry/bosh-linux-stemcell-builder](https://github.com/cloudfoundry/bosh-linux-stemcell-builder). 

This commit removes the `-jammy` (aka `-(@= stemcell.os @)`) suffix from this directory, and the resource name. The resource name change isn't necessary but does simplify the YTT somewhat.

### CODE REVIEW

**Review by lnguyen — APPROVED** [review]

**Review by jpalermo — APPROVED** [review]
👍 Still blocked on the other PR, but can be merged right after

**Pull request merged**
